### DISCUSSION TITLE
Specify Item Per Auto Crafter

### DISCUSSION THREAD

**Discussion by RobGoesVirtual** [BG-Software-LLC/WildChests#98] [Pending Feature Requests]
It would be nice to have the option to select a specific item per auto-crafter chest so that, for instance, Prismarine (using x4 Prismarine Shards) and Prismarine Bricks (using x9 Prismarine Shards) could both be active in the recipes list at the same time without interfering with each other. Thanks for your attention and for all of the amazing work you've done with these plugins. 😄

**Comment by OmerBenGera** [comment]
How would it work..?
How will the plugin tell which recipe to choose?

**Reply by RobGoesVirtual** [reply]
In a similar manner to how a storage unit knows what block to allow to pass it's filter. For instance, say that I wanted to do a chest specifically to do deepslate tiles. Just as you would start a storage unit with placing a deepslate tile in the chest to assign that id as a filter for the chest, do the same with the autocrafter to assign what block recipe to use, in this case, deepslate_tile. The functionality to check the chest for the proper ingredients to complete the recipe is already there, as is the ability to assign an item_id to a chest if I understand correctly. It would involve hooking a single recipe from the approved list that is in the config file into said chest.

In essence, instead of the autocrafter testing an incoming item_id against the item_id that was first deposited, it would test against the ids listed in the recipe in question, if it is listed in the config file.

**Reply by OmerBenGera** [reply]
You can configure which items the crafter can craft in the config. I dont understand what you are suggesting here.

**Reply by RobGoesVirtual** [reply]
If a player has two separate autocrafters setup, and a supply of sugarcane that feeds both chests at the same time, how would you have one side producing sugar and the otherside producing paper, simultaneously?

This is what is being suggested with the method above. Sorry if there was any confusion.

**Reply by OmerBenGera** [reply]
> If a player has two separate autocrafters setup, and a supply of sugarcane that feeds both chests at the same time, how would you have one side producing sugar and the otherside producing paper, simultaneously?
> 
> This is what is being suggested with the method above. Sorry if there was any confusion.

By giving him two diffenet crafters, one that can craft sugar and one that can craft paper.
There is no way for players to choose which items the crafter can craft.
